apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -329,56 +330,104 @@ public void FromDecimal(decimal input, string expectedValue)
+            PerformTest(text, handling);
+            PerformTest("-" + text, handling);
+
+            static void PerformTest(string text, LossOfPrecisionHandling handling)
+            {
+                PgNumeric numeric = PgNumeric.Parse(text);
+                Assert.Throws<OverflowException>(() => numeric.ToDecimal(handling));
+            }
         }
 
         [Theory]
-        [InlineData("123456789012345678900.123456789", "123456789012345678900.12345679")]
         [InlineData("79228162514264337593543950335.0000000001", "79228162514264337593543950335")]
-        public void ToDecimal_LossOfPrecision_Truncate(string input, string expectedValue)
+        [InlineData("0.1234567891234567891234567899999", "0.1234567891234567891234567899")]
+        [InlineData("1.1234567891234567891234567899999", "1.1234567891234567891234567899")]
+        [InlineData("123456789012345678900.123456780", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456785", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456789", "123456789012345678900.12345678")]
+        [InlineData("123456789012345678900.123456770", "123456789012345678900.12345677")]
+        [InlineData("123456789012345678900.123456775", "123456789012345678900.12345677")]
+        [InlineData("123456789012345678900.123456779", "123456789012345678900.12345677")]
+        [InlineData("7.2345678901234567890123456789", "7.2345678901234567890123456789")]
+        [InlineData("8.2345678901234567890123456789", "8.234567890123456789012345678")]
+        [InlineData("0.82345678901234567890123456789", "0.8234567890123456789012345678")]
+        [InlineData("9999999999999999999999999999.999999999999999999999999", "9999999999999999999999999999")]
+        public void ToDecimal_LossOfPrecision_Truncate(string text, string expectedDecimalText)
         {
-            // Decimal can take upto 29 digits. If input has more significant digits than decimal can handle, it will be truncated and
-            // there will be loss of precision.
-            var numeric = PgNumeric.Parse(input);
-            var convertedDecimalString = numeric.ToDecimal(LossOfPrecisionHandling.Truncate).ToString(CultureInfo.InvariantCulture);
-            Assert.Equal(expectedValue, convertedDecimalString);
+            // It's easier to do this than to vary negation in other ways.
+            PerformTest(text, expectedDecimalText);
+            PerformTest("-" + text, "-" + expectedDecimalText);
+
+            static void PerformTest(string text, string expectedDecimalText)
+            {
+                PgNumeric numeric = PgNumeric.Parse(text);
+                decimal actual = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
+                Assert.Equal(expectedDecimalText, actual.ToString(CultureInfo.InvariantCulture));
+                // Conversion via the explicit conversion should do the same thing
+                decimal actual2 = (decimal)numeric;
```

Comment:
Nit: space after cast (given that I've asked Rishabh to include that in the editorconfig :)

Reply:
(In some other tests too.)